### ISSUE TITLE
VisibilityEnabler2D will try to connect a signal multiple times when repeatedly moved inside scene tree

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#52297]
### Godot version

v3.4.beta.custom_build [a10f6022e]

### System information

W10

### Issue description

When you do `remove_child()` on VisibilityEnabler2D  and `add_child()` and again, it will try to connect some signal, but fail. It only happens when `process_parent` or `physics_process_parent` is enabled. It will throw
```
E 0:00:02.583   connect: Signal 'ready' is already connected to given method 'set_physics_process' in that object.
```

3D enabler might be affected too. Doesn't happen in 4.0.

### Steps to reproduce

1. Open reproduction project.
2. Press space a few times.

### Minimal reproduction project

[ReproductionProject.zip](https://github.com/godotengine/godot/files/7087266/ReproductionProject.zip)

**Comment by Schweini07** [comment]
With the VisibilityNotifier3D no errors show up.

I would try to tackle this problem, if possible, it shouldn't be too hard to fix.

**Comment by akien-mga** [comment]
Fixed by #53022.
